RHDEVDOCS-3777- Release Notes, New Features, and Known Issues for SBO 1.1 (#46008)
openshift/openshift-docs: modules/sbo-release-notes-1-1.adoc

For more information about the support scope of Red Hat Technology Preview features, see https://access.redhat.com/support/offerings/techpreview/.
====

[id="support-matrix-1-1_{context}"]
== Support matrix

Some features in this release are currently in Technology Preview. These experimental features are not intended for production use.

link:https://access.redhat.com/support/offerings/techpreview[Technology Preview Features Support Scope]

In the table below, features are marked with the following statuses:

- *TP*: _Technology Preview_

- *GA*: _General Availability_

Note the following scope of support on the Red Hat Customer Portal for these features:

.Support matrix
[cols="1,1",options="header"]
|===
| Feature | {servicebinding-title} 1.1
| `binding.operators.coreos.com` API group
| GA
| `servicebinding.io` API group
| TP
|===

[id="new-features-1-1_{context}"]
== New features
{servicebinding-title} 1.1 supports OpenShift Container Platform 4.10 and later running on:

* IBM Power Systems

> * IBM Z and LinuxONE

Author: Removing this.

This section highlights what is new in {servicebinding-title} 1.1:

* Service Binding Options
** Workload resource mapping: Define exactly where binding data needs to be projected for the secondary workloads.

> ** Bind workloads using a label selector

Author: Fixed now.

> [id="sbo-release-notes-1-1_{context}"]

Reviewer: ```suggestion
// Module included in the following assembly:
//
// * applications/connecting_applications_to_services/sbo-release-notes.adoc
:_content-type: REFERENCE
[id="sbo-release-notes-1-1_{context}"]
```

Reviewer: Only six of the release notes in /modules seem to have a content type attribute, but those that do have picked "REFERENCE" as the type. That seems reasonable to me.

+
[IMPORTANT]
====
*Service Binding (Spec API Tech Preview)* with the `servicebinding.io` API group is a Technology Preview feature only. Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.

For more information about the support scope of Red Hat Technology Preview features, see https://access.redhat.com/support/offerings/techpreview/.
====

[id="support-matrix-1-1_{context}"]
== Support matrix

Some features in this release are currently in Technology Preview. These experimental features are not intended for production use.

link:https://access.redhat.com/support/offerings/techpreview[Technology Preview Features Support Scope]

In the table below, features are marked with the following statuses:

- *TP*: _Technology Preview_

- *GA*: _General Availability_

Note the following scope of support on the Red Hat Customer Portal for these features:

> .Support matrix

Reviewer: Typically, a support matrices like this one cover multiple releases: a single release doesn't need a whole table - only a pair of statements like the ones on lines 8 and 9, above.  You might consider moving this table to the assembly level and including more columns for the earlier releases.

Reviewer: Maybe review how other sub-products with release notes in the /modules folder handle this question: `gitops*`, `migration*`, `op*`, and `sbo*`.

Reviewer: Disregard. I see you followed the examples in the `gitops-release-notes*.adoc` files.
This is something for discussion in the future. Don't worry about it now.

Author: Thanks for the suggestion. Have tried my best. PTAL.

[cols="1,1",options="header"]
|===
| Feature | {servicebinding-title} 1.1
| `binding.operators.coreos.com` API group
| GA
| `servicebinding.io` API group
| TP
|===

[id="new-features-1-1_{context}"]
== New features
This section highlights what is new in {servicebinding-title} 1.1:

* Service Binding Options
** Workload resource mapping: Define exactly where binding data needs to be projected for the secondary workloads.

> ** Bind new workloads using a label selector

Author: Good catch! Thank you :)

[id="fixed-issues-1-1_{context}"]
== Fixed issues
* Before this update, service bindings that used label selectors to pick up workloads did not project service binding data into the new workloads that matched the given label selectors. As a result, the Service Binding Operator could not periodically bind such new workloads. With this update, service bindings now project service binding data into the new workloads that match the given label selector. The Service Binding Operator now periodically attempts to find and bind such new workloads. link:https://issues.redhat.com/browse/APPSVC-1083[APPSVC-1083]

[id="known-issues-1-1_{context}"]
== Known issues

> * Currently, when you install the {servicebinding-title} in the single namespace installation mode, the absence of an appropriate namespace-scoped role-based access control (RBAC) rule prevents the successful binding of an application to a few known Operator-backed services that the {servicebinding-title} can automatically detect and bind to. In addition, the following error message is generated:

Reviewer: This sentence is well-crafted, but its length makes it harder to interpret. Would it be possible to divide this into two or three sentences?

Reviewer: Is there literally a "single namespace installation mode?"

Reviewer: How about:
```suggestion
* There is currently a known issue with installing {servicebinding-title} in a single namespace. The absence of an appropriate namespace-scoped role-based access control (RBAC) rule prevents {servicebinding-title} from binding an application to a few known Operator-backed services that it can detect. When this happens, it generates the following error message:
```

Author: > Is there literally a mode called "single namespace installation mode?

Yes, there is :)

Author: The **successful** word is important here, because even if the binding happens it may not be successful as the application may not yet be connected to the database service. As a result, the pod fails to start and when you try to interact with the application, it returns errors. The user is aware of this detail and hence the need for the **successful** word. That way it is more specific.

Reviewer 2: This looks accurate to me.

+
.Example error message

> ----

Author: Good catch! Thanks :)

`postgresclusters.postgres-operator.crunchydata.com "hippo" is forbidden:
        User "system:serviceaccount:my-petclinic:service-binding-operator" cannot
        get resource "postgresclusters" in API group "postgres-operator.crunchydata.com"
        in the namespace "my-petclinic"`
----
+
Workaround 1: Install the {servicebinding-title} in the `all namespaces` installation mode. As a result, the appropriate cluster-scoped RBAC rule now exists and the binding succeeds.
+

> Workaround 2: If you cannot install the {servicebinding-title} in the `all namespaces` installation mode, install the following role binding into the namespace where the {servicebinding-title} is installed: 

Reviewer: Similar questions about using the term "mode" here.

Author: Ignored :)

+
.Example: Role binding for Crunchy Postgres Operator
[source,yaml]
----
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: service-binding-crunchy-postgres-viewer
subjects:
  - kind: ServiceAccount
    name: service-binding-operator
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: service-binding-crunchy-postgres-viewer-role
----
link:https://issues.redhat.com/browse/APPSVC-1062[APPSVC-1062]

* Currently, when you modify the `ClusterWorkloadResourceMapping` resources, the {servicebinding-title} does not implement correct behavior. As a workaround, perform the following steps:
+

> . Delete any `ServiceBinding` resources using the corresponding `ClusterWorkloadResourceMapping` resource.

Reviewer: ```suggestion
. Delete any `ServiceBinding` resources that use the corresponding `ClusterWorkloadResourceMapping` resource.
```
- The "using" here was ambiguous: it could be misunderstood as "Delete any resources by using the..." rather than "Delete any resources that use the..."

. Modify the `ClusterWorkloadResourceMapping` resource.
. Re-apply the `ServiceBinding` resources that you previously removed in step 1.

> link:https://issues.redhat.com/browse/APPSVC-1102[APPSVC-1102]

Reviewer: Please have a look at your preview to see where this link appears.

> link:https://issues.redhat.com/browse/APPSVC-1102[APPSVC-1102]

Reviewer: I think you might be able to use an open block [like shown here](https://docs.asciidoctor.org/asciidoc/latest/lists/continuation/#enclose-in-open-block) to solve this formatting issue:
```suggestion
* Currently, when you modify the `ClusterWorkloadResourceMapping` resources, the {servicebinding-title} does not implement correct behavior. As a workaround, perform the following steps:
+
--
. Delete any `ServiceBinding` resources using the corresponding `ClusterWorkloadResourceMapping` resource.
. etc...etc...
--
+
link:https://issues.redhat.com/browse/APPSVC-1102[APPSVC-1102]
```

For more information about the support scope of Red Hat Technology Preview features, see https://access.redhat.com/support/offerings/techpreview/.
====

[id="support-matrix-1-1_{context}"]
== Support matrix

Some features in this release are currently in Technology Preview. These experimental features are not intended for production use.

link:https://access.redhat.com/support/offerings/techpreview[Technology Preview Features Support Scope]

In the table below, features are marked with the following statuses:

- *TP*: _Technology Preview_

- *GA*: _General Availability_

Note the following scope of support on the Red Hat Customer Portal for these features:

.Support matrix
[cols="1,1",options="header"]
|===
| Feature | {servicebinding-title} 1.1
| `binding.operators.coreos.com` API group
| GA
| `servicebinding.io` API group
| TP
|===

[id="new-features-1-1_{context}"]
== New features
This section highlights what is new in {servicebinding-title} 1.1:

* Service Binding Options
** Workload resource mapping: Define exactly where binding data needs to be projected for the secondary workloads.
** Bind new workloads using a label selector

> * Support for `ClusterWorkloadResourceMapping.servicebinding.io` resources. This support allows `ServiceBinding` resources to project the binding data into a wider variety of workloads.

Author: @sadlerap I agree. I think we can just retain L48 as it is simpler and to the point. The module anyways has the details.
We can remove L50. WDYT?

> * Support for `ClusterWorkloadResourceMapping.servicebinding.io` resources. This support allows `ServiceBinding` resources to project the binding data into a wider variety of workloads.

Author: Remoced L50
